### PR TITLE
Fix issue with arm64 noble ubuntu-server not showing correctly on KPI

### PR DESCRIPTION
**Pull request by andersson1234**

### CODE REVIEW

**Comment by andersson1234** [comment]
Weirdly enough, just running the script manually causes the right output, and shows up in the KPI. But when I run as systemd it doesn't work. Maybe my change isn't necessary but there's something I'm missing. I'll continue looking right now.

**Comment by andersson1234** [comment]
ADT-433

File: metrics/collectors/images/images_collector.py [hunk 1]
```diff
@@ -36,6 +36,8 @@ def rsync_list_images(self):
         rsync = ""
         for url in RSYNC_SERVER_REQUESTS:
             for img in IMAGE_FORMATS:
+                # stopping annoying traceback as ubuntustudio has no .img.xz
+                # Though I don't think this is the source of the bug
```
Comment (paride) [inline metrics/collectors/images/images_collector.py:40]: to be dropped, as discussed.

File: metrics/collectors/images/images_collector.py [hunk 2]
```diff
@@ -113,6 +114,7 @@ def collect(self):
                             "release": series,
                             "current_or_pending": current_or_pending,
                             "architecture": arch,
+                            "preinstalled": preinstalled,
```
Comment (paride) [inline metrics/collectors/images/images_collector.py:117]: I asked the release team, and apparently "preinstalled" is best referred to as an image "type". So I think we should add a "type" tag and set it to "live", "preinstalled", and similar. (Can we easily identify an image type for all the images?)

Comment (andersson1234) [reply metrics/collectors/images/images_collector.py]: Amended, please re-review

**Review by paride — CHANGES_REQUESTED** [review]
we want "preinstalled", "live" as an image "type".

**Comment by andersson1234** [comment]
> we want "preinstalled", "live" as an image "type".

Amended, please re-review

File: metrics/collectors/images/images_collector.py [hunk 3]
```diff
@@ -92,6 +93,7 @@ def collect(self):
                 # the path ends with the image filename
                 image_name = path_table[-1]
 
+                image_type = "preinstalled" if ".img.xz" in path else "live"
```
Comment (paride) [inline metrics/collectors/images/images_collector.py:96]: the "type" info has to come from the path, see the code above.

- "daily-preinstalled" should also have flavor=ubuntu
- "daily-live" -> type = live
- "daily-preinstalled" => type = preinstalled
- by default: type = None

File: metrics/collectors/images/images_collector.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
     "rsync://cdimage.ubuntu.com/cdimage/*/*/dvd/*/*",
 ]
 IMAGE_FORMATS = [".iso", ".img.xz"]
+IMAGE_TYPE = "pre-installed"
```
Comment (paride) [inline metrics/collectors/images/images_collector.py:19]: don't add unused vars

**Review by paride — CHANGES_REQUESTED** [review]

**Comment by andersson1234** [comment]
amended, please re-review

**Comment by paride** [comment]
please also fix the flavor, as I mentioned in my previous comment

**Comment by paride** [comment]
See line 86. I think preinstalled images are not getting the flavor set right, correct?

**Review by paride — APPROVED** [review]
LGTM, thanks!

**Comment by andersson1234** [comment]
I pushed again, but just to fix CI :)

File: metrics/collectors/images/images_collector.py [hunk 5]
```diff
@@ -83,7 +83,9 @@ def collect(self):
                 elif path_table[0] in self.active_series:
                     flavor = "ubuntu"
                 # or with 'daily-live'
-                elif path_table[0] == "daily-live":
+                elif (
+                    path_table[0] == "daily-live" or path_table == "daily-preinstalled"
```
Comment (murraybd) [inline metrics/collectors/images/images_collector.py:87]: Why is it path_table[0] in part and then path_table?

File: metrics/collectors/images/images_collector.py [hunk 6]
```diff
@@ -92,6 +94,11 @@ def collect(self):
                 # the path ends with the image filename
                 image_name = path_table[-1]
 
+                image_type = None
+                if "daily-preinstalled" in path:
+                    image_type = "preinstalled"
+                elif "daily-live" in path:
+                    image_type = "live"
```
Comment (murraybd) [inline metrics/collectors/images/images_collector.py:101]: Checking for these strings anywhere in path seems fragile to me. Could we use a more specific check e.g. something like path[99]=='daily-preinstalled'?

**Review by murraybd — COMMENTED** [review]
I've a couple of questions regarding the approach here.

**Comment by andersson1234** [comment]
amended, please re-review

**Comment by andersson1234** [comment]
I am actually going to amend this mp to fix this issue as well:
https://github.com/ubuntu/ubuntu-release-metrics/issues/30

**Comment by andersson1234** [comment]
Latest commit fixes the xubuntu issue i linked before:
```
 {'fields': {'age': 0, 'size': 1696036864},
  'measurement': 'daily_image_details',
  'tags': {'architecture': 'amd64',
           'current_or_pending': 'current',
           'flavor': 'xubuntu',
           'image_type': 'minimal',
           'release': 'noble'}},
```

File: metrics/collectors/images/images_collector.py [hunk 7]
```diff
@@ -92,6 +95,12 @@ def collect(self):
                 # the path ends with the image filename
                 image_name = path_table[-1]
 
+                image_type = None
+                for part_of_path in path_table:
+                    # If statement like this to cover daily-live, daily-legacy
+                    # daily-minimal and daily-preinstalled
+                    if "daily-" in part_of_path:
+                        image_type = part_of_path.replace("daily-", "")
```
Comment (paride) [inline metrics/collectors/images/images_collector.py:103]: You can `break` here.

Looking at
```
RSYNC_SERVER_REQUESTS = [
    "rsync://cdimage.ubuntu.com/cdimage/daily*/*/*",
    "rsync://cdimage.ubuntu.com/cdimage/*/daily*/*/*",
    "rsync://cdimage.ubuntu.com/cdimage/*/dvd/*/*",
    "rsync://cdimage.ubuntu.com/cdimage/*/*/daily*/*/*",
    "rsync://cdimage.ubuntu.com/cdimage/*/*/dvd/*/*",
]
```
looks like the image "types" can be `daily*` or `dvd` probably deprecated. However if we want it more general we could do:
```
if part_of_path.startswith("daily") or part_of_path == "dvd":
    image_type = part_or_path
    break
```
this means keeping the `daily-` bit. Given that we also have "types" without "daily-", this is probably fine after all.

**Comment by andersson1234** [comment]
Amended, please re-review

File: metrics/collectors/images/images_collector.py [hunk 8]
```diff
@@ -92,6 +95,18 @@ def collect(self):
...
+                    # If statement like this to cover daily-live, daily-legacy
+                    # daily-minimal, daily-preinstalled and dvd
+                    starts_with_daily = part_of_path.startswith("daily-")
+                    if starts_with_daily or part_of_path == "dvd":
+                        image_type = (
+                            part_of_path.replace("daily-", "")
+                            if starts_with_daily
+                            else part_of_path
+                        )
+                        break
```
Comment (paride) [inline metrics/collectors/images/images_collector.py:109]: So you want to keep the `daily-`, fine, but why can't we keep it simple?
```
for part_of_path in path_table:
    if part_of_path.startswith("daily-") or part_of_path == "dvd":
        image_type = part_or_path.replace("daily-", "")
        break
```

**Review by paride — APPROVED** [review]
lgtm, but I didn't test it

**Review by murraybd — APPROVED** [review]

**Pull request merged**